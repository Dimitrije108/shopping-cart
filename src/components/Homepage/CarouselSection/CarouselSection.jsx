import { useState } from "react";
import { Link } from "react-router-dom";
import styles from "./CarouselSection.module.css";

const slides = [
	{
		src: "/src/assets/star-destroyer.jpg",
		alt: "Star Destroyer Capital ship",
		marketingText: "Only a Sith Deals in Full Retail. Shop Our Discounts!",
		linkTo: "/starships/capital",
		buttonText: "Best Capital Ships in the Galaxy - Shop Now!",
	},
	{
		src: "/src/assets/slave-i.jpg",
		alt: "Slave I starship",
		marketingText: "Bounty Hunting Is a Complicated Profession — Get the Right Ship!",
		linkTo: "/starships/starfighter",
		buttonText: "Unleash Your Inner Pilot — Buy a Starfighter!",
	},
	{
		src: "/src/assets/imperial-shuttle.jpg",
		alt: "Imperial Shuttle Transport ship",
		marketingText: "More Than a Ship — A Lifestyle.",
		linkTo: "/starships/transport",
		buttonText: "Shop Elegance — Shop Our Transport Starships!",
	},
];

export default function CarouselSection() {
	const [index, setIndex] = useState(0);
	// Change to the next carousel slide
	const prevSlide = () => {
		setIndex((prev) => Math.max(prev - 1, 0));
	};
	// Change to the previous carousel slide
	const nextSlide = () => {
		setIndex((prev) => Math.min(prev + 1, slides.length - 1));
	};
	// Change carousel slide to the cliked one
	const changeSlide = (index) => {
		setIndex(index);
	};

	return (
		<section 
			className={styles.carouselSection}
			id="toscroll"
		>
			<div className={styles.carouselCont}>
				<div 
					className={styles.carouselTrack} 
					style={{ transform: `translateX(-${index * 100}vw)` }}
				>
					{slides.map((slide, i) => (
						<div key={i} className={styles.slideCont}>
							<img src={slide.src} alt={slide.alt} />
							<div className={styles.carouselMarketingText}>
								<h2 className={styles.standardText}>
									{slide.marketingText}
								</h2>
								<div className={styles.marketingPositioningCont}>
									<div>
										<p 
											className={styles.imperial}
											title="Imperial"
										>
											{slide.marketingText}
										</p>
										<p 
											className={styles.naboo} 
											title="Naboo"
										>
											{slide.marketingText}
										</p>
									</div>
									<Link to={slide.linkTo}>
										<button className={styles.shopCapitalBtn}>
											{slide.buttonText}
										</button>
									</Link>
								</div>
							</div>
						</div>	
					))}
				</div>
				<div className={styles.carouselQuickNav}>
					<button 
						className={styles.carouselQuickNavBtn} 
						onClick={() => changeSlide(0)}
					>
						<div 
							className={`
								${styles.navSquare} 
								${index === 0 ? styles.active : ""}
							`}
						></div>
					</button>
					<button 
						className={styles.carouselQuickNavBtn} 
						onClick={() => changeSlide(1)}
					>
						<div
							className={`
								${styles.navSquare} 
								${index === 1 ? styles.active : ""}
							`}
						></div>
					</button>
					<button 
						className={styles.carouselQuickNavBtn} 
						onClick={() => changeSlide(2)}
					>
						<div
							className={`
								${styles.navSquare} 
								${index === 2 ? styles.active : ""}
							`}
						></div>
					</button>
				</div>
			</div>
			<button 
				className={styles.leftArrow}
				onClick={prevSlide}
			>
				<img 
					src="/src/assets/carousel-left-arrow.svg" 
					alt="left arrow" 
					width={30}
				/>
			</button>
			<button 
				className={styles.rightArrow}
				onClick={nextSlide}
			>
				<img 
					src="/src/assets/carousel-right-arrow.svg" 
					alt="right arrow" 
					width={30}
				/>
			</button>
		</section>
	)
};
