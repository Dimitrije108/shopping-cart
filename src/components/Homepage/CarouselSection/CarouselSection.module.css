.carouselSection {
	max-height: calc(100vh - 60px);

	border-top: 1px solid rgb(214, 214, 214);
	border-bottom: 1px solid rgb(214, 214, 214);

	display: flex;
	overflow: hidden;
  align-items: center;

	position: relative;
}

.carouselCont {
	min-width: 100vw;

	display: flex;
	flex: 1;
  overflow: hidden;
}

.carouselTrack {
	width: calc(300vw - 42px);
	transition: transform 0.5s ease-in-out;
	display: flex;
}

.slideCont {
	flex: 0 0 calc(100vw - 14px);
	height: auto;
	max-height: calc(100vh - 60px);
	position: relative;
}

.slideCont picture {
  display: block;
  width: 100%;
  height: 100%;
  max-height: calc(100vh - 60px);
}

.slideCont img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center center;
}

.leftArrow,
.rightArrow {
	min-height: 100%;

	background: transparent;
	border: none;
	cursor: pointer;

	transition: all 0.5s ease-in-out;

	position: absolute;
}

.rightArrow {
	right: 0;
}

.leftArrow:hover {
	background: linear-gradient(to left, transparent, rgb(10, 10, 10));
	border: none;
}

.rightArrow:hover {
	background: linear-gradient(to right, transparent, rgb(10, 10, 10));
	border: none;
}

.leftArrow > img,
.rightArrow > img {
	width: 30px;
}

.carouselMarketingText {
	max-width: 1250px;
	width: 100%;
	padding: 1rem;

	background-color: rgba(0, 0, 0, 0.6);

	position: absolute;
	bottom: 4rem;
  left: 50%;
  transform: translateX(-50%);
}

.standardText {
	font-size: 2.5rem;
	font-family: 'DINNext bold';
}

.imperial {
	font-family: Imperial;
	font-size: 1rem;
}

.naboo {
	font-family: Naboo;
	font-size: 1rem;
}

.shopCapitalBtn {
	margin-top: 1rem;
	padding: 1rem 3rem;

	font-family: 'DINNext bold';
	font-size: 1.5rem;
	letter-spacing: 1px;
	
  border: 2px solid rgb(179, 179, 179);

	transition: all 0.3s ease-in-out;
}

.shopCapitalBtn:hover {
  border: 2px solid rgb(238, 238, 238);
	letter-spacing: 2px;
	background-color: rgba(0, 0, 0, 0.2);
}

.carouselQuickNav {
	position: absolute;
	bottom: 1rem;
  left: 50%;
  transform: translateX(-50%);
}

.carouselQuickNavBtn {
	margin: 0 12px;
	padding: 0;
	border: 1px solid rgb(189, 189, 189);
}

.carouselQuickNavBtn:hover {
	border: 1px solid rgb(243, 243, 243);
}

.navSquare {
	height: 4px;
	width: 2.5rem;
}

.navSquare.active {
	background-color: rgb(243, 243, 243);
}

@media (max-width: 1250px) {
	.standardText {
		font-size: 2rem;
	}

	.imperial {
		font-size: 0.9rem;
	}
	
	.naboo {
		font-size: 0.9rem;
	}
}

@media (max-width: 700px) {
	.standardText {
		font-size: 1.8rem;
	}

	.imperial {
		font-size: 0.8rem;
	}
	
	.naboo {
		font-size: 0.8rem;
	}

	.shopCapitalBtn {
		padding: 1rem;
		letter-spacing: 0;
	}
	
	.shopCapitalBtn:hover {
		letter-spacing: 0.5px;
	}
}

@media (max-width: 600px) {
	.imperial {
		font-size: 0.7rem;
	}
	
	.naboo {
		font-size: 0.7rem;
	}

	.shopCapitalBtn {
		padding: 1rem;
		font-size: 1.3rem;
	}

	.leftArrow > img,
	.rightArrow > img {
		width: 25px;
	}
}

@media (max-width: 500px) {

	.standardText {
		font-size: 1.4rem;
	}

	.imperial {
		font-size: 8px;
	}
	
	.naboo {
		font-size: 8px;
	}

	.shopCapitalBtn {
		padding: 0.5rem;
		font-size: 1.1rem;
		letter-spacing: 0;
	}
	
	.shopCapitalBtn:hover {
		letter-spacing: 0.5px;
	}

	.leftArrow > img,
	.rightArrow > img {
		width: 20px;
	}

	.carouselQuickNavBtn {
		margin: 0 6px;
	}
	
	.navSquare {
		height: 3px;
		width: 2rem;
	}
}
