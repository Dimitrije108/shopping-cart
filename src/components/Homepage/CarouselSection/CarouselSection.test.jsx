import { describe, it, expect } from "vitest";
import { render, screen  } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import CarouselSection from "./CarouselSection";

describe("Homepage's Carousel Section component", () => {
	it("Carousel Section snapshot", () => {
		const carouselSection = render(
			<MemoryRouter>
				<CarouselSection />
			</MemoryRouter>
		)

		expect(carouselSection).toMatchSnapshot();
	});

	describe("Carousel Section elements render correctly", () => {
		it("Carousel Track elements render correctly", () => {
			render(
				<MemoryRouter>
					<CarouselSection />
				</MemoryRouter>
			)
			
			const firstSlideImg = screen.getByAltText("Star Destroyer Capital ship");
			const firstSlideHeading = screen.getByRole("heading", { name: "Only a Sith Deals in Full Retail. Shop Our Discounts!" });
			const firstSlideBtn = screen.getByRole("button", { name: "Best Capital Ships in the Galaxy - Shop Now!" });

			const buttonLinks = screen.getAllByTestId("starshipsLink");
			const paragraphs = screen.getAllByTitle("Imperial");
			
			expect(firstSlideImg).toHaveAttribute("src", "/src/assets/images/carousel/star-destroyer.jpg");
			expect(firstSlideHeading).toBeInTheDocument();
			expect(firstSlideBtn).toBeInTheDocument();

			expect(buttonLinks[0]).toHaveAttribute("href", "/starships/capital#capitalSection");
			expect(paragraphs[0]).toHaveTextContent("Only a Sith Deals in Full Retail. Shop Our Discounts!");
		});

		it("Carousel quick nav renders correctly", () => {
			render(
				<MemoryRouter>
					<CarouselSection />
				</MemoryRouter>
			)
			
			const buttonLinks = screen.getAllByTestId("quickNavBtn");

			expect(buttonLinks[2]).toBeInTheDocument();
			expect(buttonLinks).toHaveLength(3);
		});

		it("Carousel previous and next buttons render correctly", () => {
			render(
				<MemoryRouter>
					<CarouselSection />
				</MemoryRouter>
			)
			
			const leftBtn = screen.getByTestId("leftBtn");
			const rightBtn = screen.getByTestId("rightBtn");
			const leftIcon = screen.getByAltText("left arrow");
			const rightIcon = screen.getByAltText("right arrow");

			expect(leftBtn).toBeInTheDocument();
			expect(rightBtn).toBeInTheDocument();
			expect(leftIcon).toHaveAttribute("src", "/src/assets/icons/carousel-left-arrow.svg");
			expect(rightIcon).toHaveAttribute("src", "/src/assets/icons/carousel-right-arrow.svg");
		});
	});

	describe("Carousel Section functionality", () => {
		it("Carousel next slide functionality works correctly", async () => {
			const user = userEvent.setup();

			render(
				<MemoryRouter>
					<CarouselSection />
				</MemoryRouter>
			)
			
			const rightBtn = screen.getByTestId("rightBtn");
			const slides = screen.getAllByTestId("carouselSlide");

			expect(slides[0]).toHaveAttribute("aria-hidden", "false");
			expect(slides[1]).toHaveAttribute("aria-hidden", "true");
			expect(slides[2]).toHaveAttribute("aria-hidden", "true");

			await user.click(rightBtn);

			expect(slides[0]).toHaveAttribute("aria-hidden", "true");
			expect(slides[1]).toHaveAttribute("aria-hidden", "false");
			expect(slides[2]).toHaveAttribute("aria-hidden", "true");

			await user.click(rightBtn);
			await user.click(rightBtn);
			await user.click(rightBtn);

			expect(slides[0]).toHaveAttribute("aria-hidden", "true");
			expect(slides[1]).toHaveAttribute("aria-hidden", "true");
			expect(slides[2]).toHaveAttribute("aria-hidden", "false");
		});

		it("Carousel previous slide functionality works correctly", async () => {
			const user = userEvent.setup();

			render(
				<MemoryRouter>
					<CarouselSection />
				</MemoryRouter>
			)
			
			const leftBtn = screen.getByTestId("leftBtn");
			const rightBtn = screen.getByTestId("rightBtn");
			const slides = screen.getAllByTestId("carouselSlide");

			expect(slides[0]).toHaveAttribute("aria-hidden", "false");
			expect(slides[1]).toHaveAttribute("aria-hidden", "true");
			expect(slides[2]).toHaveAttribute("aria-hidden", "true");

			await user.click(rightBtn);
			await user.click(rightBtn);
			await user.click(leftBtn);

			expect(slides[0]).toHaveAttribute("aria-hidden", "true");
			expect(slides[1]).toHaveAttribute("aria-hidden", "false");
			expect(slides[2]).toHaveAttribute("aria-hidden", "true");

			await user.click(leftBtn);
			await user.click(leftBtn);
			await user.click(leftBtn);

			expect(slides[0]).toHaveAttribute("aria-hidden", "false");
			expect(slides[1]).toHaveAttribute("aria-hidden", "true");
			expect(slides[2]).toHaveAttribute("aria-hidden", "true");
		});

		it("Carousel change slide functionality works correctly", async () => {
			const user = userEvent.setup();

			render(
				<MemoryRouter>
					<CarouselSection />
				</MemoryRouter>
			)
			
			const quickNacBtns = screen.getAllByTestId("quickNavBtn");
			const slides = screen.getAllByTestId("carouselSlide");

			expect(slides[0]).toHaveAttribute("aria-hidden", "false");
			expect(slides[1]).toHaveAttribute("aria-hidden", "true");
			expect(slides[2]).toHaveAttribute("aria-hidden", "true");

			await user.click(quickNacBtns[2]);

			expect(slides[0]).toHaveAttribute("aria-hidden", "true");
			expect(slides[1]).toHaveAttribute("aria-hidden", "true");
			expect(slides[2]).toHaveAttribute("aria-hidden", "false");

			await user.click(quickNacBtns[1]);

			expect(slides[0]).toHaveAttribute("aria-hidden", "true");
			expect(slides[1]).toHaveAttribute("aria-hidden", "false");
			expect(slides[2]).toHaveAttribute("aria-hidden", "true");
		});
	});
});
