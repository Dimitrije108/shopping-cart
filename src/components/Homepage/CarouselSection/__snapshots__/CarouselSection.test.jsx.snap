// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage's Carousel Section component > Carousel Section snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="_carouselSection_d6bb1e"
        id="toscroll"
      >
        <div
          class="_carouselCont_d6bb1e"
        >
          <div
            class="_carouselTrack_d6bb1e"
            style="transform: translateX(-0vw);"
          >
            <div
              aria-hidden="false"
              class="_slideCont_d6bb1e"
              data-testid="carouselSlide"
            >
              <img
                alt="Star Destroyer Capital ship"
                src="/src/assets/star-destroyer.jpg"
              />
              <div
                class="_carouselMarketingText_d6bb1e"
              >
                <h2
                  class="_standardText_d6bb1e"
                >
                  Only a Sith Deals in Full Retail. Shop Our Discounts!
                </h2>
                <div
                  class="_marketingPositioningCont_d6bb1e"
                >
                  <div>
                    <p
                      class="_imperial_d6bb1e"
                      title="Imperial"
                    >
                      Only a Sith Deals in Full Retail. Shop Our Discounts!
                    </p>
                    <p
                      class="_naboo_d6bb1e"
                      title="Naboo"
                    >
                      Only a Sith Deals in Full Retail. Shop Our Discounts!
                    </p>
                  </div>
                  <a
                    data-discover="true"
                    data-testid="storeLink"
                    href="/starships/capital"
                  >
                    <button
                      class="_shopCapitalBtn_d6bb1e"
                    >
                      Best Capital Ships in the Galaxy - Shop Now!
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="_slideCont_d6bb1e"
              data-testid="carouselSlide"
            >
              <img
                alt="Slave I starship"
                src="/src/assets/slave-i.jpg"
              />
              <div
                class="_carouselMarketingText_d6bb1e"
              >
                <h2
                  class="_standardText_d6bb1e"
                >
                  Bounty Hunting Is a Complicated Profession — Get the Right Ship!
                </h2>
                <div
                  class="_marketingPositioningCont_d6bb1e"
                >
                  <div>
                    <p
                      class="_imperial_d6bb1e"
                      title="Imperial"
                    >
                      Bounty Hunting Is a Complicated Profession — Get the Right Ship!
                    </p>
                    <p
                      class="_naboo_d6bb1e"
                      title="Naboo"
                    >
                      Bounty Hunting Is a Complicated Profession — Get the Right Ship!
                    </p>
                  </div>
                  <a
                    data-discover="true"
                    data-testid="storeLink"
                    href="/starships/starfighter"
                  >
                    <button
                      class="_shopCapitalBtn_d6bb1e"
                    >
                      Unleash Your Inner Pilot — Buy a Starfighter!
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="_slideCont_d6bb1e"
              data-testid="carouselSlide"
            >
              <img
                alt="Imperial Shuttle Transport ship"
                src="/src/assets/imperial-shuttle.jpg"
              />
              <div
                class="_carouselMarketingText_d6bb1e"
              >
                <h2
                  class="_standardText_d6bb1e"
                >
                  More Than a Ship — A Lifestyle.
                </h2>
                <div
                  class="_marketingPositioningCont_d6bb1e"
                >
                  <div>
                    <p
                      class="_imperial_d6bb1e"
                      title="Imperial"
                    >
                      More Than a Ship — A Lifestyle.
                    </p>
                    <p
                      class="_naboo_d6bb1e"
                      title="Naboo"
                    >
                      More Than a Ship — A Lifestyle.
                    </p>
                  </div>
                  <a
                    data-discover="true"
                    data-testid="storeLink"
                    href="/starships/transport"
                  >
                    <button
                      class="_shopCapitalBtn_d6bb1e"
                    >
                      Shop Elegance — Shop Our Transport Starships!
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="_carouselQuickNav_d6bb1e"
          >
            <button
              class="_carouselQuickNavBtn_d6bb1e"
              data-testid="quickNavBtn"
            >
              <div
                class="
								_navSquare_d6bb1e 
								_active_d6bb1e
							"
              />
            </button>
            <button
              class="_carouselQuickNavBtn_d6bb1e"
              data-testid="quickNavBtn"
            >
              <div
                class="
								_navSquare_d6bb1e 
								
							"
              />
            </button>
            <button
              class="_carouselQuickNavBtn_d6bb1e"
              data-testid="quickNavBtn"
            >
              <div
                class="
								_navSquare_d6bb1e 
								
							"
              />
            </button>
          </div>
        </div>
        <button
          class="_leftArrow_d6bb1e"
          data-testid="leftBtn"
        >
          <img
            alt="left arrow"
            src="/src/assets/carousel-left-arrow.svg"
            width="30"
          />
        </button>
        <button
          class="_rightArrow_d6bb1e"
          data-testid="rightBtn"
        >
          <img
            alt="right arrow"
            src="/src/assets/carousel-right-arrow.svg"
            width="30"
          />
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="_carouselSection_d6bb1e"
      id="toscroll"
    >
      <div
        class="_carouselCont_d6bb1e"
      >
        <div
          class="_carouselTrack_d6bb1e"
          style="transform: translateX(-0vw);"
        >
          <div
            aria-hidden="false"
            class="_slideCont_d6bb1e"
            data-testid="carouselSlide"
          >
            <img
              alt="Star Destroyer Capital ship"
              src="/src/assets/star-destroyer.jpg"
            />
            <div
              class="_carouselMarketingText_d6bb1e"
            >
              <h2
                class="_standardText_d6bb1e"
              >
                Only a Sith Deals in Full Retail. Shop Our Discounts!
              </h2>
              <div
                class="_marketingPositioningCont_d6bb1e"
              >
                <div>
                  <p
                    class="_imperial_d6bb1e"
                    title="Imperial"
                  >
                    Only a Sith Deals in Full Retail. Shop Our Discounts!
                  </p>
                  <p
                    class="_naboo_d6bb1e"
                    title="Naboo"
                  >
                    Only a Sith Deals in Full Retail. Shop Our Discounts!
                  </p>
                </div>
                <a
                  data-discover="true"
                  data-testid="storeLink"
                  href="/starships/capital"
                >
                  <button
                    class="_shopCapitalBtn_d6bb1e"
                  >
                    Best Capital Ships in the Galaxy - Shop Now!
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="_slideCont_d6bb1e"
            data-testid="carouselSlide"
          >
            <img
              alt="Slave I starship"
              src="/src/assets/slave-i.jpg"
            />
            <div
              class="_carouselMarketingText_d6bb1e"
            >
              <h2
                class="_standardText_d6bb1e"
              >
                Bounty Hunting Is a Complicated Profession — Get the Right Ship!
              </h2>
              <div
                class="_marketingPositioningCont_d6bb1e"
              >
                <div>
                  <p
                    class="_imperial_d6bb1e"
                    title="Imperial"
                  >
                    Bounty Hunting Is a Complicated Profession — Get the Right Ship!
                  </p>
                  <p
                    class="_naboo_d6bb1e"
                    title="Naboo"
                  >
                    Bounty Hunting Is a Complicated Profession — Get the Right Ship!
                  </p>
                </div>
                <a
                  data-discover="true"
                  data-testid="storeLink"
                  href="/starships/starfighter"
                >
                  <button
                    class="_shopCapitalBtn_d6bb1e"
                  >
                    Unleash Your Inner Pilot — Buy a Starfighter!
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="_slideCont_d6bb1e"
            data-testid="carouselSlide"
          >
            <img
              alt="Imperial Shuttle Transport ship"
              src="/src/assets/imperial-shuttle.jpg"
            />
            <div
              class="_carouselMarketingText_d6bb1e"
            >
              <h2
                class="_standardText_d6bb1e"
              >
                More Than a Ship — A Lifestyle.
              </h2>
              <div
                class="_marketingPositioningCont_d6bb1e"
              >
                <div>
                  <p
                    class="_imperial_d6bb1e"
                    title="Imperial"
                  >
                    More Than a Ship — A Lifestyle.
                  </p>
                  <p
                    class="_naboo_d6bb1e"
                    title="Naboo"
                  >
                    More Than a Ship — A Lifestyle.
                  </p>
                </div>
                <a
                  data-discover="true"
                  data-testid="storeLink"
                  href="/starships/transport"
                >
                  <button
                    class="_shopCapitalBtn_d6bb1e"
                  >
                    Shop Elegance — Shop Our Transport Starships!
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="_carouselQuickNav_d6bb1e"
        >
          <button
            class="_carouselQuickNavBtn_d6bb1e"
            data-testid="quickNavBtn"
          >
            <div
              class="
								_navSquare_d6bb1e 
								_active_d6bb1e
							"
            />
          </button>
          <button
            class="_carouselQuickNavBtn_d6bb1e"
            data-testid="quickNavBtn"
          >
            <div
              class="
								_navSquare_d6bb1e 
								
							"
            />
          </button>
          <button
            class="_carouselQuickNavBtn_d6bb1e"
            data-testid="quickNavBtn"
          >
            <div
              class="
								_navSquare_d6bb1e 
								
							"
            />
          </button>
        </div>
      </div>
      <button
        class="_leftArrow_d6bb1e"
        data-testid="leftBtn"
      >
        <img
          alt="left arrow"
          src="/src/assets/carousel-left-arrow.svg"
          width="30"
        />
      </button>
      <button
        class="_rightArrow_d6bb1e"
        data-testid="rightBtn"
      >
        <img
          alt="right arrow"
          src="/src/assets/carousel-right-arrow.svg"
          width="30"
        />
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
